Add operators.json entry for Clojure

jeff303 opened the pull request:
## What GitHub issue does this PR apply to?

No issue opened, just working to add more complete Clojure support.

## What changed and why?

Adding operators support for Clojure

## Checklist

   <!-- 
   Each - [ ] below is a checkbox that will show up on the PR.
   Either add an X inside the [X], or submit the PR and click the checkboxes.
   -->

- [N/A] I claimed any associated issue(s) and they are not someone else's
- [X] I have looked at documentation to ensure I made any revisions correctly
- [X] I tested my changes locally to ensure they work
- [N/A] (If editing Django) I have added or edited any appropriate unit tests for my changes


## Any additional comments or things to be aware of while reviewing?

jeff303 committed:
Add operators.json entry for Clojure

jeff303 committed:
Add a few missing bitwise functions

geekygirlsarah commented on web/thesauruses/clojure/1/operators.json:
```diff
@@ -0,0 +1,163 @@
+    "percentage": {
+      "code": "%",
+      "name": "Percentage operator",
+      "not-implemented": "true"
```
Is this % or not implemented? Can't have both.

jeff303 replied:
Sorry, in this case I actually did check the rendered website (which was correct) but forgot to remove it from source).  Should be fixed now.

geekygirlsarah committed:
Merge branch 'main' into clojure-add-operators

Commit:
PR feedback

Add more unsupported operators

Add some entries that were supported but missing

geekygirlsarah commented on web/thesauruses/clojure/1/operators.json:
```diff
@@ -0,0 +1,223 @@
+    },
+    "addition_assignment": {
+      "comment": "Clojure doesn't have the traditional notion of assignment as in procedural languages, but it can apply in some situations (ex: atoms and volatile refs).",
+      "code": "(let [x (volatile! 0)]\n  (doseq [addend (range 10)]\n    (vswap! x + addend))\n  @x)",
```
This is basically inventing the solution, and Code Thesaurus is a reference tool for what exists. So therefore it should probably be "not-implemented". (Same for similar assignments below)

geekygirlsarah commented on web/thesauruses/clojure/1/operators.json:
```diff
@@ -0,0 +1,223 @@
+    },
+    "exponential": {
+      "comment": "No built in function, but easy enough to write.",
+      "code": "(defn exp [x n]\n  (reduce * (repeat n x)))",
```
Same for these, don't write new functions, just report if the language presently can do it. (so "not-implemented")

geekygirlsarah commented on web/thesauruses/clojure/1/operators.json:
```diff
@@ -0,0 +1,223 @@
+    },
+    "absolute_value": {
+      "comment": "Using the built-in JDK method.",
+      "code": "(Math/abs -4)",
```
Importing library functions still counts, so this is fine.

geekygirlsarah reviewed (changes requested):
If you can revise for the few spots you wrote similar functionality and just show it's not built-in, I can merge this in.

geekygirlsarah committed:
Merge branch 'main' into clojure-add-operators

Commit:
PR feedback: marking things not-supported

geekygirlsarah committed:
Merge branch 'main' into clojure-add-operators

geekygirlsarah commented on web/thesauruses/clojure/1/operators.json:
```diff
@@ -0,0 +1,213 @@
+    "factorial": {
+      "comment": "No built in function, but easy enough to write.",
+      "code": "(defn ! [n]\n  (reduce *' (range 1 (inc n))))",
+      "name": "Factorial operator"
```
My first thought is this is a custom-implementation and shouldn't be here. If it's not built-in, it's `not-implemented: true`. 

Second, I always hate using things like "this is easy" because you don't know the experience level of the person using the tool. It's "easy" to someone like me and you, but might not be to someone newer to Clojure or programming.

jeff303 replied:
Done.

geekygirlsarah commented on web/thesauruses/clojure/1/operators.json:
```diff
@@ -0,0 +1,213 @@
+    },
+    "ternary": {
+      "comment": "Use the `if` function to provide a ternary operator equivalent.",
+      "code": "(if (> 2 1) \"Yes\" \"No\")",
```
This is also inventing a oslution. Should be `not-implemented` too.

jeff303 replied:
I actually think in this case it truly is equivalent.  Since every expression returns a value, this is how you would accomplish the same thing that requires the operator in other languages.  See [here](https://reshmeeauckloo.wordpress.com/2015/12/13/clojure-if-cond-case/) for a better explanation than I can give.

Nonetheless I don't feel that strongly about it, so I marked not-implemented.

geekygirlsarah replied:
I think I also misread it because when I look at it now, it clearly works like a ternary. I can either merge this in or you can roll this back.

geekygirlsarah reviewed (dismissed):
Make the two things `not-implemented` and I'll merge it.

geekygirlsarah reviewed (changes requested):
Make the two things not-implemented and I'll merge it.

jeff303 committed:
Marking operators not-implemented

geekygirlsarah committed:
Merge branch 'main' into clojure-add-operators

geekygirlsarah commented:
Alright, I'll merge it in since you fixed the not-implemented part but I didn't hear back on the ternary. (We can always revert it back some other time if you want.)

Thanks again!
